### PR TITLE
Redeliver failed webhooks to account for temporary errors

### PR DESCRIPTION
**Pull request by kalikiana**
See: https://progress.opensuse.org/issues/165144

### CODE REVIEW

**Comment by kalikiana** [comment]
This is an alternative to #5876 which I found after preparing the draft based on the boilerplate from official GitHub docs, and probably what we would prefer as we don't need to maintain this in the repo **plus** it already supports multiple hooks out of the box.

File: .github/workflows/redeliver.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+permissions:
+  contents: read
+
+jobs:
+  redeliver:
+  - name: Redeliver failed webhooks
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: gateixeira/retrigger-webhook-action@main
```
Comment (perlpunk) [inline .github/workflows/redeliver.yaml:20]: This action doesn't seem to be an official github action.
Also I think we should never use the `main` version of a third party action we don't control.
And also the usage
https://github.com/gateixeira/retrigger-webhook-action
seems to be totally different. It should not be a scheduled workflow on its own. It says "Add the following snippet to an existing workflow file".
And I think this should really be tested somehow, maybe you can add a dummy workflow in your fork.

File: .github/workflows/redeliver.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      env:
+        token: ${{ secrets.GH_TOKEN_FOR_ACTIONS }}
+        owner: ${{ github.repository_owner }}
+        repo: ${{ github.event.repository.name }}
+        last_redelivery_variable_name: 'LAST_WEBHOOK_REDELIVERY'
```
Comment (perlpunk) [inline .github/workflows/redeliver.yaml:25]: Where does that come from? I don't really understand the documentation of the action.

Comment (perlpunk) [reply .github/workflows/redeliver.yaml:25]: Also, which webhook should it attempt to redeliver? we have more than one and probably need to give it an id, e.g. like documented `webhook_id: 'YOUR_HOOK_ID'`

File: .github/workflows/redeliver.yaml (hunk 1 above)
Comment (kalikiana) [reply .github/workflows/redeliver.yaml:20]: The alternative is #5876 where we implement it from scratch.

I agree on adding a version, though.

File: .github/workflows/redeliver.yaml (hunk 2 above)
Comment (kalikiana) [reply .github/workflows/redeliver.yaml:25]: Ah. I forgot to add the comment here. This is to store the last timestamp in a variable.

File: .github/workflows/redeliver.yaml (hunk 1 above)
Comment (perlpunk) [reply .github/workflows/redeliver.yaml:20]: > What existing workflow would this be if not this one? I read **existing** as any workflow where this action is useful, and the schedule is what GitHub docs recommend.

I don't know. I don't find the documentation of that action very useful and also not very reassuring :)

**Comment by kalikiana** [comment]
> Did you have a look at the code this executes? I haven't and thus cannot really approve this PR. As mentioned in the other PR we should probably test what implications this can have for the various webhooks we have. (We don't want things to happen twice or e.g. re-trigger CI checks based on an old state.)

https://github.com/gateixeira/retrigger-webhook-action/blob/f606bae35fe73db18b10fe1ce6b5e52a734b6062/src/main.ts

It is based on [the officially documented boilerplate](https://docs.github.com/en/webhooks/using-webhooks/automatically-redelivering-failed-deliveries-for-a-repository-webhook) and uses the same pattern to avoid retriggering hooks more than once.

File: .github/workflows/redeliver.yaml (hunk 1 above)
Comment (perlpunk) [reply .github/workflows/redeliver.yaml]: I never said that I don't like the idea.

And for a proof of concept we still need the proof :)

**Comment by okurz** [comment]
With https://github.com/openSUSE/open-build-service/pull/16834 this shouldn't be necessary anymore, right?

**Comment by kalikiana** [comment]
> With [openSUSE/open-build-service#16834](https://github.com/openSUSE/open-build-service/pull/16834) this shouldn't be necessary anymore, right?

Ack.

**Pull request closed without merging**